### PR TITLE
test(logging): add profiling test

### PR DESCRIPTION
**Pull request by Yongxuanzhang**

### CODE REVIEW

File: tests/unit/logging/test_profile.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    yield
+    default_logger.logger.propagate = False
+
+
+def test_logging_profile_profiling(caplog, default_logger_propagate):
+    @profiling
+    def foo():
+        print(1)
+    foo()
+    assert "memory" in caplog.text
```
Comment (JoanFM) [inline tests/unit/logging/test_profile.py:21]: please use single quote

File: tests/unit/logging/test_profile.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+def default_logger_propagate():
+    default_logger.logger.propagate = True
+    yield
+    default_logger.logger.propagate = False
+
+
+def test_logging_profile_profiling(caplog, default_logger_propagate):
+    @profiling
+    def foo():
+        print(1)
```
Comment (JoanFM) [inline tests/unit/logging/test_profile.py:19]: we could allocate some memory here or sleep for some time to showcase how it is used and how the memory and some time is returned.

Tests tend to be good way to clarify behavior

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: tests/unit/logging/test_profile.py (hunk 2 above)
Comment (Yongxuanzhang) [reply tests/unit/logging/test_profile.py]: Sure! I add some changes. Not sure if this is ok

**Comment by hanxiao** [comment]
for some reason, merging this PR breaks the `test_logging` ci, reverting now

**Pull request merged**